src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using System.Collections.Immutable;
+
+#nullable enable
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public class RequestExecutionQueue<RequestContextType> : IRequestExecutionQueue<RequestContextType>
```

Comment:
Does this need to be public? There isn't much here that is protected, perhaps the IRequestContextFactory could come from a service that is passed in, and then this could be a private implementation detail, as the default queue that CLaSP provides. Extenders can use a different queue if they want, but don't necessarily need access to the default impl?

Reply:
Hmm, makes sense. I'll give it a shot and see how it goes.